FileBufferTest.testPersist flaky

**Summary**

- How often does the test fail? Top3 in CW 27
- Does it block your work? no
- Do we suspect that it is a real failure? unknown

**Failures**

Currently no further information available.

Failure:
```
java.lang.RuntimeException: java.io.FileNotFoundException: target/test-files/da16110c-2c15-4fc1-b4ce-9c66edba3366 (No such file or directory)
	at io.atomix.storage.buffer.FileBytes.<init>(FileBytes.java:73)
	at io.atomix.storage.buffer.FileBuffer.allocate(FileBuffer.java:121)
	at io.atomix.storage.buffer.FileBuffer.allocate(FileBuffer.java:76)
	at io.atomix.storage.buffer.FileBufferTest.testPersist(FileBufferTest.java:66)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeLazy(JUnitCoreWrapper.java:119)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:87)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:157)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:428)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:562)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:548)
Caused by: java.io.FileNotFoundException: target/test-files/da16110c-2c15-4fc1-b4ce-9c66edba3366 (No such file or directory)
	at java.base/java.io.RandomAccessFile.open0(Native Method)
	at java.base/java.io.RandomAccessFile.open(RandomAccessFile.java:345)
	at java.base/java.io.RandomAccessFile.<init>(RandomAccessFile.java:259)
	at java.base/java.io.RandomAccessFile.<init>(RandomAccessFile.java:214)
	at io.atomix.storage.buffer.FileBytes.<init>(FileBytes.java:68)
	... 43 more
```

--- a/atomix/storage/pom.xml
+++ b/atomix/storage/pom.xml
@@ -14,7 +14,9 @@
   ~ See the License for the specific language governing permissions and
   ~ limitations under the License.
   -->
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <artifactId>atomix-storage</artifactId>
 
   <modelVersion>4.0.0</modelVersion>
@@ -43,6 +45,13 @@
       <artifactId>kryo</artifactId>
       <groupId>com.esotericsoftware</groupId>
     </dependency>
+
+    <dependency>
+      <groupId>io.zeebe</groupId>
+      <artifactId>zeebe-test-util</artifactId>
+      <scope>test</scope>
+    </dependency>
+
     <dependency>
       <artifactId>junit</artifactId>
       <groupId>junit</groupId>

--- a/atomix/storage/src/test/java/io/atomix/storage/buffer/FileTesting.java
+++ b/atomix/storage/src/test/java/io/atomix/storage/buffer/FileTesting.java
@@ -16,6 +16,10 @@
  */
 package io.atomix.storage.buffer;
 
+import static io.zeebe.test.util.TestEnvironment.getTestForkNumber;
+import static io.zeebe.test.util.TestEnvironment.getTestMavenId;
+import static java.lang.String.format;
+
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.FileVisitResult;
@@ -27,14 +31,23 @@ import java.nio.file.attribute.BasicFileAttributes;
 import java.util.UUID;
 
 public abstract class FileTesting {
+
+  private static final String getTestRunPathElement() {
+    return format("fork-%d_mavenID-%d/", getTestForkNumber(), getTestMavenId());
+  }
+
+  private static final String getFolderNameForTestData() {
+    return "target/test-files/" + getTestRunPathElement();
+  }
+
   public static File createFile() {
-    final File file = new File("target/test-files/" + UUID.randomUUID().toString());
+    final File file = new File(getFolderNameForTestData() + UUID.randomUUID().toString());
     file.getParentFile().mkdirs();
     return file;
   }
 
   public static void cleanFiles() {
-    final Path directory = Paths.get("target/test-files/");
+    final Path directory = Paths.get(getFolderNameForTestData());
     try {
       Files.walkFileTree(
           directory,

--- a/test-util/src/main/java/io/zeebe/test/util/TestEnvironment.java
+++ b/test-util/src/main/java/io/zeebe/test/util/TestEnvironment.java
@@ -1,0 +1,54 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.test.util;
+
+import io.zeebe.util.ZbLogger;
+import org.slf4j.Logger;
+
+public class TestEnvironment {
+  private static final Logger LOG = new ZbLogger("io.zeebe.test.util");
+
+  private static final String TEST_FORK_NUMBER_PROPERTY_NAME = "testForkNumber";
+  private static final String TEST_MAVEN_ID_PROPERTY_NAME = "testMavenId";
+
+  public static int getTestForkNumber() {
+    int testForkNumber = 0;
+    try {
+      final String testForkNumberProperty = System.getProperty(TEST_FORK_NUMBER_PROPERTY_NAME);
+      if (testForkNumberProperty != null) {
+        testForkNumber = Integer.valueOf(testForkNumberProperty);
+      } else {
+        LOG.warn(
+            "No system property '{}' set, using default value {}",
+            TEST_FORK_NUMBER_PROPERTY_NAME,
+            testForkNumber);
+      }
+    } catch (final Exception e) {
+      LOG.warn("Failed to read test fork number system property", e);
+    }
+    return testForkNumber;
+  }
+
+  public static int getTestMavenId() {
+    int testMavenId = 0;
+    try {
+      final String testMavenIdProperty = System.getProperty(TEST_MAVEN_ID_PROPERTY_NAME);
+      if (testMavenIdProperty != null) {
+        testMavenId = Integer.valueOf(testMavenIdProperty);
+      } else {
+        LOG.warn(
+            "No system property '{}' set, using default value {}",
+            TEST_MAVEN_ID_PROPERTY_NAME,
+            testMavenId);
+      }
+    } catch (final Exception e) {
+      LOG.warn("Failed to read test maven id system property", e);
+    }
+    return testMavenId;
+  }
+}

--- a/test-util/src/main/java/io/zeebe/test/util/socket/SocketUtil.java
+++ b/test-util/src/main/java/io/zeebe/test/util/socket/SocketUtil.java
@@ -7,15 +7,13 @@
  */
 package io.zeebe.test.util.socket;
 
+import io.zeebe.test.util.TestEnvironment;
 import io.zeebe.util.ZbLogger;
 import java.net.InetSocketAddress;
 import org.slf4j.Logger;
 
 public final class SocketUtil {
   static final Logger LOG = new ZbLogger("io.zeebe.test.util.SocketUtil");
-
-  private static final String TEST_FORK_NUMBER_PROPERTY_NAME = "testForkNumber";
-  private static final String TEST_MAVEN_ID_PROPERTY_NAME = "testMavenId";
 
   private static final String DEFAULT_HOST = "localhost";
   private static final int BASE_PORT = 25600;
@@ -25,8 +23,8 @@ public final class SocketUtil {
   private static final PortRange PORT_RANGE;
 
   static {
-    final int testForkNumber = getTestForkNumber();
-    final int testMavenId = getTestMavenId();
+    final int testForkNumber = TestEnvironment.getTestForkNumber();
+    final int testMavenId = TestEnvironment.getTestMavenId();
 
     LOG.info(
         "Starting socket assignment with testForkNumber {} and testMavenId {}",
@@ -49,41 +47,5 @@ public final class SocketUtil {
 
   public static InetSocketAddress getNextAddress() {
     return PORT_RANGE.next();
-  }
-
-  private static int getTestForkNumber() {
-    int testForkNumber = 0;
-    try {
-      final String testForkNumberProperty = System.getProperty(TEST_FORK_NUMBER_PROPERTY_NAME);
-      if (testForkNumberProperty != null) {
-        testForkNumber = Integer.valueOf(testForkNumberProperty);
-      } else {
-        LOG.warn(
-            "No system property '{}' set, using default value {}",
-            TEST_FORK_NUMBER_PROPERTY_NAME,
-            testForkNumber);
-      }
-    } catch (final Exception e) {
-      LOG.warn("Failed to read test fork number system property", e);
-    }
-    return testForkNumber;
-  }
-
-  private static int getTestMavenId() {
-    int testMavenId = 0;
-    try {
-      final String testMavenIdProperty = System.getProperty(TEST_MAVEN_ID_PROPERTY_NAME);
-      if (testMavenIdProperty != null) {
-        testMavenId = Integer.valueOf(testMavenIdProperty);
-      } else {
-        LOG.warn(
-            "No system property '{}' set, using default value {}",
-            TEST_MAVEN_ID_PROPERTY_NAME,
-            testMavenId);
-      }
-    } catch (final Exception e) {
-      LOG.warn("Failed to read test maven id system property", e);
-    }
-    return testMavenId;
   }
 }

